Sources/TuistKit/Services/InitService.swift

```diff
@@ -92,23 +98,42 @@ class InitService {
 
         let directories = try templatesDirectoryLocator.templateDirectories(at: path)
         if let templateName = templateName {
-            guard
-                let templateDirectory = directories.first(where: { $0.basename == templateName })
-            else { throw InitServiceError.templateNotFound(templateName) }
-            let template = try templateLoader.loadTemplate(at: templateDirectory)
-            let parsedAttributes = try parseAttributes(
-                name: name,
-                platform: platform,
-                requiredTemplateOptions: requiredTemplateOptions,
-                optionalTemplateOptions: optionalTemplateOptions,
-                template: template
-            )
+            var template: Template?
+            if templateName.isURL {
+                try fileHandler.inTemporaryDirectory { temporaryDirectory in
+                    let templateDirectoryName = temporaryDirectory.appending(.init("\(templateName)")).url.lastPathComponent
```

Comment:
You can use `appending(component: String)` instead `.appending(.init())`

Reply:
this whole variable also seems redundant - why not just use `templateName` directly?